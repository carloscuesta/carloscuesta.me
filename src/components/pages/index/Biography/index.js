// @flow
import React from 'react'

import Wrapper from 'src/components/shared/Wrapper'
import styles from './styles.module.css'

const Biography = () => (
  <section className={styles.biography}>
    <Wrapper>
      <div className='row middle-xs'>
        <div className='col-xs-12 col-sm-6'>
          <h3 className={styles.biographyText}>
            I'm a 24 years old <b>Front End Engineer</b> based in <b>Barcelona</b>,
            working at <a href='https://ulabox.com' rel='noopener noreferrer'><b>Ulabox</b></a>.
            I'm a coding passionate that is focused on <b>JavaScript</b>, specifically in React & React-Native.
            Interested in design, tech and <b>OpenSource</b> ❤️
          </h3>

          <h3 className={styles.biographyText}>
            You can find me on{' '}
            <a href='https://github.com/carloscuesta' target='_blank' rel='noopener noreferrer'>GitHub</a>,{' '}
            <a href='https://twitter.com/crloscuesta' target='_blank' rel='noopener noreferrer'>Twitter</a>{' and '}
            <a href='https://linkedin.com/in/crloscuesta' target='_blank' rel='noopener noreferrer'>LinkedIn</a>
          </h3>
        </div>

        <div className='col-xs-12 col-sm-6'>
          <img
            alt='Carlos Cuesta'
            className={styles.profilePicture}
            src='images/carloscuesta.jpg'
          />
        </div>
      </div>
    </Wrapper>

  </section>
)

export default Biography
