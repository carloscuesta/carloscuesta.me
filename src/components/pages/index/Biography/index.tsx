import Wrapper from 'src/components/shared/Wrapper'
import styles from './styles.module.css'

const Biography = () => (
  <section>
    <Wrapper>
      <div className={styles.container}>
        <div className={styles.biographyContainer}>
          <p className={styles.text}>
            I&#39;m a Front End Engineer interested in{' '}
            <b>design</b>, <b>under engineering</b> and <b>open source</b>.
            I work at <a target='_blank' rel='noopener noreferrer' href='https://n26.com'>N26</a>.
          </p>

          <p className={styles.text}>
            I&#39;m focused on building products with JavaScript, specifically React and React-Native.
          </p>

          <p className={styles.text}>
            When I&#39;m not coding, I&#39;m likely at the gym or surfing the city with my longboard.
          </p>

          <p className={styles.text}>
            You can find me on{' '}
            <a href='https://github.com/carloscuesta' target='_blank' rel='noopener noreferrer'>GitHub</a>,{' '}
            <a href='https://twitter.com/crloscuesta' target='_blank' rel='noopener noreferrer'>Twitter</a>{' and '}
            <a href='https://linkedin.com/in/crloscuesta' target='_blank' rel='noopener noreferrer'>LinkedIn</a>.
          </p>
        </div>

        <div className={styles.pictureContainer}>
          <img
            alt='Carlos Cuesta'
            className={`lazyload ${styles.picture}`}
            data-src='https://res.cloudinary.com/carloscuesta/image/upload/s--ekQN9GP_--/c_scale,w_320/v1594588508/carloscuesta.jpg'
            src='https://res.cloudinary.com/carloscuesta/image/upload/s--rq7ZUVpY--/c_scale,e_blur:100,h_120,q_10,w_120/v1594588508/carloscuesta.jpg'
          />
        </div>

      </div>
    </Wrapper>
  </section>
)

export default Biography
