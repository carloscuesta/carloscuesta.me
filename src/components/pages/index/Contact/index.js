// @flow
import type { Element } from 'react'

import Wrapper from 'src/components/shared/Wrapper'
import SectionTitle from 'src/components/shared/SectionTitle'
import styles from './styles.module.css'

const Contact = (): Element<'section'> => (
  <section>
    <Wrapper>
      <SectionTitle
        subTitle="Let's make something together."
        title='Contact'
      />

      <div className={styles.wrapper}>
        <p className={styles.text}>
          Feel free to send me an email if you&#39;re looking for a front end engineer,
          have a question or just want to say hi! üôã‚Äç‚ôÇÔ∏è
        </p>

        <p className={styles.text}>
          <a
            className={styles.button}
            href='mailto:hi@carloscuesta.me'
            rel='noopener noreferrer'
            target='_blank'
          >
            Say hello
          </a>
        </p>
      </div>
    </Wrapper>
  </section>
)

export default Contact
