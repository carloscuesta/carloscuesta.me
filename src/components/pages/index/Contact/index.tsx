import Wrapper from 'src/components/shared/Wrapper'
import SectionTitle from 'src/components/shared/SectionTitle'

const Contact = () => (
  <section>
    <Wrapper>
      <SectionTitle
        subTitle="Let's make something together."
        title='Contact'
      />

      <div>
        <p>
          Feel free to send me an email if you&#39;re looking for a front end engineer,
          have a question or just want to say hi! 🙋‍♂️
        </p>

        <p>
          <a
            href='mailto:hi@carloscuesta.me'
            rel='noopener noreferrer'
            target='_blank'
          >
            Say hello
          </a>
        </p>
      </div>
    </Wrapper>
  </section>
)

export default Contact
