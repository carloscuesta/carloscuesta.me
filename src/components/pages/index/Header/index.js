// @flow
import type { Element } from 'react'
import Link from 'next/link'

import Wrapper from 'src/components/shared/Wrapper'
import styles from './styles.module.css'

const Header = (): Element<'header'> => (
  <header className={styles.container}>
    <Wrapper>
      <h1 className={styles.title}>Hi, I&#39;m <span className={styles.name}>Carlos Cuesta</span></h1>
      <span className={styles.wave}>👋🏼</span>
      <p className={styles.bio}>
        I&#39;m a <b>Front End Engineer</b> from Barcelona.
        I <a target='_blank' rel='noopener noreferrer' href='https://github.com/carloscuesta'>code</a>, <Link href='/blog'><a>write</a></Link> and build stuff on internet.
      </p>
    </Wrapper>
  </header>
)

export default Header
