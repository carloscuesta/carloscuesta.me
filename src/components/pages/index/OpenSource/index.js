// @flow
import React, { type Element } from 'react'

import { type Repository } from 'src/utils/api/github/mutators'
import Wrapper from 'src/components/shared/Wrapper'
import SectionTitle from 'src/components/shared/SectionTitle'
import Project from './Project'

type Props = {
  repositories: Array<Repository>
}

const OpenSource = (props: Props): Element<'section'> => (
  <section>
    <Wrapper>
      <SectionTitle
        title='Open Source'
        subTitle='Featured open source projects.'
      />

      <div className='row'>
        {props.repositories.map((repository) => (
          <Project repository={repository} key={repository.name} />
        ))}
      </div>
    </Wrapper>
  </section>
)

export default OpenSource
