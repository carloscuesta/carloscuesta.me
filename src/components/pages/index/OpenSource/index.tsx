import { type Repository } from 'src/utils/api/github/mutators'
import Wrapper from 'src/components/shared/Wrapper'
import SectionTitle from 'src/components/shared/SectionTitle'
import Project from './Project'

type Props = {
  repositories: Repository[]
}

const OpenSource = (props: Props) => (
  <section>
    <Wrapper>
      <SectionTitle
        title='Open Source'
        subTitle='Featured open source projects.'
        viewAllLink='/opensource'
      />

      <div className='row'>
        {props.repositories.map((repository) => (
          <Project repository={repository} key={repository.name} />
        ))}
      </div>
    </Wrapper>
  </section>
)

export default OpenSource
