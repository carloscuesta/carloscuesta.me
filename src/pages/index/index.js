// @flow
import React from 'react'
import { NextSeo } from 'next-seo'

import { fetchPosts } from 'src/utils/api/blog'
import { fetchRepositories } from 'src/utils/api/github'
import { type Post } from 'src/utils/api/blog/mutators'
import { type Repository } from 'src/utils/api/github/mutators'
import Header from 'src/components/pages/index/Header'
import Biography from 'src/components/pages/index/Biography'
import Writings from 'src/components/pages/index/Writings'
import OpenSource from 'src/components/pages/index/OpenSource'

type Props = {
  posts: Array<Post>,
  repositories: Array<Repository>
}

const Index = (props: Props) => (
  <>
    <NextSeo canonical='https://carloscuesta.me' />
    <Header />
    <main>
      <Biography />
      <Writings posts={props.posts} />
      <OpenSource repositories={props.repositories} />
    </main>
  </>
)

export const getStaticProps = async (): Promise<{ props: Props }> => {
  const [posts, repositories] = await Promise.all([
    fetchPosts(),
    fetchRepositories()
  ])

  return {
    props: {
      posts: posts.slice(0, 6),
      repositories
    }
  }
}

export default Index
