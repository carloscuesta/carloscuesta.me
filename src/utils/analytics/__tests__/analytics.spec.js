import * as analytics from '../index'
import * as stubs from './stubs'

describe('analytics', () => {
  it('should match analytics module', () => {
    expect(analytics).toMatchSnapshot()
  })

  describe('trackPageView', () => {
    const gtag = jest.fn()

    beforeAll(() => {
      window.gtag = gtag
    })

    afterAll(() => {
      window.gtag = undefined
    })

    it('should call window.gtag with ID and page_path arguments', () => {
      analytics.trackPageView(stubs.url)

      expect(gtag).toHaveBeenCalledWith(
        'config',
        analytics.GA_TRACKING_ID,
        { page_path: stubs.url }
      )
    })
  })
})
