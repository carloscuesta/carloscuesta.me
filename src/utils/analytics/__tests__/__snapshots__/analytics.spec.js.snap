// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analytics should match analytics module 1`] = `
Object {
  "GA_TRACKING_ID": "UA-67824860-3",
  "trackPageView": [Function],
}
`;
